worker/src/main/java/org/apache/celeborn/service/deploy/worker/storage/MapDataPartitionReader.java

```diff
@@ -436,7 +436,11 @@ public void release() {
     // we can safely release if reader reaches error or (read/send finished)
     synchronized (lock) {
       if (!isReleased) {
-        logger.debug("release reader for stream {}", this.streamId);
+        logger.debug("release reader for stream {}", streamId);
+        // tell client that this stream is finished.
+        if (readFinished) {
+          associatedChannel.writeAndFlush(new BufferStreamEnd(streamId));
```

Comment:
Will this be called in time?

Reply:
after both read and send data  are finished , it will be called